server/mailet/dkim/src/test/java/org/apache/james/jdkim/mailets/ConvertTo7BitTest.java

```diff
@@ -0,0 +1,242 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jdkim.mailets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import javax.mail.Address;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.james.util.MimeMessageUtil;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import com.github.fge.lambdas.Throwing;
+
+class ConvertTo7BitTest {
+
+    private static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding";
+    private static final String X_MIME_AUTOCONVERTED = "X-MIME-Autoconverted";
+    private static final String QUOTED_PRINTABLE = "quoted-printable";
+    private static final String MESSAGE_BODY_8BIT = "A 8bit encoded body with \u20ACuro symbol.";
+    private static final String MESSAGE_BODY_QUOTED_PRINTABLE = "A 8bit encoded body with =E2=82=ACuro symbol.";
+    private static final String EIGHT_BIT = "8bit";
+    private static final String BASE64 = "base64";
+
+    private ConvertTo7Bit testee;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        testee = new ConvertTo7Bit();
+        testee.init(FakeMailetConfig.builder()
+            .build());
+    }
+
+    @Nested
+    class When7BitMail {
+        @Nested
+        class WhenTextContent {
+            @Test
+            void serviceShouldKeepMessageUnTouch() throws Exception {
+                MimeMessage mimeMessageBefore = mimeMessageQuotedPrintable();
+                Mail mail = FakeMail.builder()
+                    .name("a-mail-with-quoted-printable-encoding")
+                    .mimeMessage(mimeMessageBefore)
+                    .build();
+
+                testee.service(mail);
+                MimeMessage mimeMessageAfter = mail.getMessage();
+
+                assertThat(mimeMessageContent(mimeMessageAfter))
```

Comment:
are you sure that the "before" is not changed by the mailet?

Reply:
oh yeah, right!